bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/CheckpointSourceList.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.bookie;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.google.common.collect.Lists;
+
+public class CheckpointSourceList implements CheckpointSource {
+
+    private final List<? extends CheckpointSource> checkpointSourcesList;
+
+    public CheckpointSourceList(List<? extends CheckpointSource> checkpointSourcesList) {
+        this.checkpointSourcesList = checkpointSourcesList;
+    }
+
+    @Override
+    public Checkpoint newCheckpoint() {
+        return new CheckpointList(this);
+    }
+
+    @Override
+    public void checkpointComplete(Checkpoint checkpoint, boolean compact) throws IOException {
+        if (checkpoint == Checkpoint.MAX || checkpoint == Checkpoint.MIN) {
+            return;
+        }
+
+        checkArgument(checkpoint instanceof CheckpointList);
+        CheckpointList checkpointList = (CheckpointList) checkpoint;
+
+        checkArgument(checkpointList.source == this);
+        checkpointList.checkpointComplete(compact);
+    }
+
+    private static class CheckpointList implements Checkpoint {
+        private final CheckpointSourceList source;
+        private final List<Checkpoint> checkpoints;
+
+        public CheckpointList(CheckpointSourceList source) {
+            this.source = source;
+            this.checkpoints = Lists.newArrayListWithCapacity(source.checkpointSourcesList.size());
+            for (CheckpointSource checkpointSource : source.checkpointSourcesList) {
+                checkpoints.add(checkpointSource.newCheckpoint());
+            }
+        }
+
+        private void checkpointComplete(boolean compact) throws IOException {
```

Comment:
The `compact` was already present in the `Checkpoint` interface. According to the javadoc: 

```
     * @param compact
     *          Flag to compact old checkpoints.
```

The point of `CheckPointSourcesList` is to act as a meta-checkpoint source, delegating to all the individual sources, so we just need to forward the `compact` parameter that we get passed on.